### PR TITLE
CT-831 - Correct egress check to pass groups named "default" by exception

### PR DESCRIPTION
**Pull request by danjoneslf**
..and update unit tests.

### CODE REVIEW

File: chalice/tests/chalicelib/criteria/test_aws_ec2_egress_restriction.py [hunk 1]
```diff
@@ -105,3 +105,14 @@ def test_evaluate_fail(self):
             # tests
             output = self._evaluate_invariant_assertions(event, item, whitelist)
             self._evaluate_failed_status_assertions(item, output)
+
+    def test_evaluate_pass_by_exception(self):
+        """
```
Comment (0atman) [inline chalice/tests/chalicelib/criteria/test_aws_ec2_egress_restriction.py]: why not docstring?

File: chalice/tests/chalicelib/criteria/test_aws_ec2_egress_restriction.py [hunk 2]
```diff
@@ -105,3 +105,14 @@ def test_evaluate_fail(self):
...
+        """
+        # input params
+        event = {}
+        whitelist = []
+        for item in EGRESS_RESTRICTION["pass-by-exception"]["SecurityGroups"]:
+            # tests
+            output = self._evaluate_invariant_assertions(event, item, whitelist)
```
Comment (0atman) [inline chalice/tests/chalicelib/criteria/test_aws_ec2_egress_restriction.py]: bad name, `output`

File: chalice/tests/chalicelib/criteria/test_aws_ec2_egress_restriction.py [hunk 3]
```diff
@@ -105,3 +105,14 @@ def test_evaluate_fail(self):
...
             output = self._evaluate_invariant_assertions(event, item, whitelist)
             self._evaluate_failed_status_assertions(item, output)
+
+    def test_evaluate_pass_by_exception(self):
+        """
+        """
+        # input params
+        event = {}
+        whitelist = []
+        for item in EGRESS_RESTRICTION["pass-by-exception"]["SecurityGroups"]:
```
Comment (0atman) [inline chalice/tests/chalicelib/criteria/test_aws_ec2_egress_restriction.py]: very bad name `item`

File: chalice/tests/chalicelib/criteria/test_aws_ec2_egress_restriction.py (hunk 2 above)
Comment (0atman) [inline chalice/tests/chalicelib/criteria/test_aws_ec2_egress_restriction.py]: bad name `output`

File: chalice/tests/chalicelib/criteria/test_aws_ec2_egress_restriction.py (hunk 3 above)
Comment (0atman) [inline chalice/tests/chalicelib/criteria/test_aws_ec2_egress_restriction.py]: very bad name `item`

**Comment by danjoneslf** [comment]
Don't disagree with any of that but do we want to do all that refactoring under this ticket. There's little value in making that change just for this one test.

**Comment by 0atman** [comment]
Agreed. I thought the modified lines were all-new.

**Pull request merged**